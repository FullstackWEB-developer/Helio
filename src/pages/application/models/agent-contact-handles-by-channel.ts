import {ChannelTypes} from '@shared/models';

export interface AgentContactHandlesByChannel {
    agent: string;
    channel: ChannelTypes;
    startInterval?: Date;
    endInterval?: Date;
    contactsHandled?: number
}
