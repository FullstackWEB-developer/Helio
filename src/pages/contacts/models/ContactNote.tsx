export interface ContactNote {
    id?: number;
    noteText: string;
    createdOn?: Date;
    createdBy?: string;
}