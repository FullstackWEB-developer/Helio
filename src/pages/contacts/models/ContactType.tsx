export enum ContactType {
    Individual = 'Individual',
    Company = 'Company'
}