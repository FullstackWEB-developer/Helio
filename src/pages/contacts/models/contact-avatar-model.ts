import {Icon} from '@components/svg-icon/icon';
export interface ContactAvatarModel {
    initials?: string;
    className?: string; 
    isCompany?: boolean;
    iconType?: Icon;
    iconClassName?: string;
    iconFillClass?: string;
}