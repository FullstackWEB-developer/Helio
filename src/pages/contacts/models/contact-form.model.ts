import {Option} from "@components/option/option";

export interface ContactFormModel {
    id?: string;
    firstName?: string;
    lastName?: string;
    companyName?: string;
    category: number | Option;
    jobTitle?: string;
    department?: string;
    email?: string;
    workMainPhone?: string;
    workMainExtension?: string;
    workDirectPhone?: string;
    mobile?: string;
    fax?: string;
    website?: string;
    primaryAddressLine?: string;
    primaryApt?: string;
    primaryCity?: string;
    primaryState?: string | Option;
    primaryZipCode?: string;
    shippingAddressLine?: string;
    shippingApt?: string;
    shippingCity?: string;
    shippingState?: string | Option;
    shippingZipCode?: string;
    billingAddressLine?: string;
    billingApt?: string;
    billingCity?: string;
    billingState?: string | Option;
    billingZipCode?: string;
    isStarred?: boolean;
    relatedId?: string;
}