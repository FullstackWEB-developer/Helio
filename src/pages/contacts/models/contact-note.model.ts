export interface ContactNote {
    id?: string;
    noteText: string;
    createdOn: Date;
    createdByName: string;
    contactId: string;
}
export interface AddContactNoteProps {
    contactId: string;
    contactNoteDto: ContactNote
}