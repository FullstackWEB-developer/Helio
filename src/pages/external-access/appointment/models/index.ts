export * from './appointment-type.model';
export * from './appointment-slot.model';
export * from './appointment.model';
export * from './appointment-cancellation.model';
export * from './appointment-cancel-reason.model';
export * from './appointment-slot-request.model';
