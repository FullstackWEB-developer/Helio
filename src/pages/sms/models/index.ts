export * from "./sms-new-message-steps";
export * from "./sms-notification-data";
export * from "./sms-query-type";
