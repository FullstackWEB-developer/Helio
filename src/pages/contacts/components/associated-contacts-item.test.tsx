import i18n from '../../../i18nForTests';
import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import customParseFormat from 'dayjs/plugin/customParseFormat';
import {unmountComponentAtNode} from 'react-dom';
import {fireEvent, render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import AssociatedContactsItem from './associated-contacts-item';
import Router from "react-router-dom";
import { AssociatedContact, ContactType } from '@shared/models';
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useParams: jest.fn(),
   }));
describe("AddNewContact tests", () => {
    let container: HTMLDivElement | null;
    const mockState = {
        emailState: {
            unreadEmails: 0,
            messageSummaries: []
        },
        appUserState: {
            liveAgentStatuses: [],
            appUserDetails: {
                id: ""
            }
        },
        lookupsState: {

        },
        ticketState: {
            lookupValues: []
        },
        appState: {
            smsTemplates: [],
            emailTemplates: []
        }
    };

    beforeEach(async () => {
        await i18n.init();
        dayjs.extend(duration);
        dayjs.extend(utc);
        dayjs.extend(customParseFormat);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders AddNewContact correctly", async () => {
        let contact = {
            companyName: "",
            firstName: "",
            id: "123",
            lastName: "",
            type: ContactType.Company
        } as AssociatedContact;
        jest.spyOn(Router, 'useParams').mockReturnValue({ id: '1234' })
        const {asFragment, getByTestId} = render(<TestWrapper mockState={mockState}>
            <AssociatedContactsItem contact={contact}/>
        </TestWrapper>);
        fireEvent.click(getByTestId(contact.id));
        expect(asFragment()).toMatchSnapshot();
    });
})
