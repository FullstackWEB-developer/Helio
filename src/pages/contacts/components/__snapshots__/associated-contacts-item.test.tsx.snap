// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNewContact tests renders AddNewContact correctly 1`] = `
<DocumentFragment>
  <div
    class="associated-contacts-item h-10 border-b contact-border-color-lighter body2 grid grid-cols-4 items-center"
  >
    <span>
       
    </span>
    <span />
    <div
      class="justify-self-end pr-6 cursor-pointer"
    >
      <div
        class=""
      >
        <svg
          class="icon-medium "
          data-testid="123"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(-1555 -621)"
          >
            <rect
              fill="none"
              height="24"
              transform="translate(1555 621)"
              width="24"
            />
            <path
              class="contact-light-fill"
              d="M12,4.5A11.827,11.827,0,0,0,1,12a11.817,11.817,0,0,0,22,0A11.827,11.827,0,0,0,12,4.5ZM12,17a5,5,0,1,1,5-5A5,5,0,0,1,12,17Zm0-8a3,3,0,1,0,3,3A3,3,0,0,0,12,9Z"
              transform="translate(1555 620.5)"
            />
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
